python_modules/dagster/dagster/_core/definitions/declarative_scheduling/legacy/legacy_context.py

```diff
@@ -250,18 +247,50 @@ def materialized_since_previous_tick_subset(self) -> ValidAssetSubset:
             ),
         )
 
+    @property
+    @root_property
+    def _previous_tick_discarded_subset(self) -> Optional[AssetSubset]:
+        """Fetches the unique id corresponding to the DiscardOnMaxMaterializationsExceededRule, if
+        that rule is part of the broader condition.
+        """
+        from ...auto_materialize_rule_impls import DiscardOnMaxMaterializationsExceededRule
+        from ..operators import NotAssetCondition
+        from .rule_condition import RuleCondition
+
+        if len(self.condition.children) != 3:
+            return None
```

Comment:
this feels a little brittle - at least with no context. At least warrants a comment i think